enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1050 @@
+    * Use `priorityClassName: hypershift-control-plane` for the driver operator.
+    * Phase 2: add metrics services + TLS keys.
+
+* **aws-ebs-csi-driver-operator**:
+  * Add a new cmdline option for the guest cluster kubeconfig file location.
+    * Parse both kubeconfigs:
+      * One from projected service account, which leads to the management cluster.
+      * Second from the new cmdline option introduced above. This one leads to the guest cluster.
+    * Rework CSI driver Deployment to run only the control plane parts of the CSI driver, i.e.:
+      * Remove `hostNetwork: true`.
+      * Add `controller` as the first command line arg.
+      * Set env. var. `AWS_REGION=us-east2` with the actual guest cluster region. The region can be read from
+        `Infrastructure` CR.
+      * This removes dependency on AWS metadata service and allows us to run the controller parts of the driver outside
+        of AWS.
+    * Rework CSI driver DaemonSet to run only node parts of the CSI driver.
+      * Add `node` as the first command line arg.
+      * This allows us not to give the DaemonSet any AWS credentials.
+  * Only on HyperShift:
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * Replace namespaces in the driver Deployment, PDBs and other objects that are created in the management cluster.
+      They must be created in the same namespace as the operator.
+    * Update the CSI driver control-plane Deployment: 
+      * Give it kubeconfig to the guest cluster as a Secret volume.
+      * Pass it Secret with STS creds generated by control-plane-operator.
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Phase 2: add metrics services + TLS keys.
+
+* **AWS EBS CSI driver**:
+  * Phase 1: no code changes.
+    * Controller parts will run in the management cluster, but they get kubeconfig to the guest cluster.
+    * Node DaemonSet will get the default projected kubeconfig that can access objects only in the
+    openshift-cluster-csi-drivers namespace.
+  * Phase 2 (or even later): reduce memory requirements of AWS EBS CSI driver control-plane Deployment.
+    * Review memory requirements of each sidecar.
+    * Union kube-rbac-proxies, if possible.
+    * Union all sidecars into a single container, so they share informers? This will need serious investment upstream.
+  * Phase 2 (or even later): move leader-election of CSI sidecars to the management cluster.
+    * I.e. the sidecars get a special kubeconfig just for the leader election.
+    * The sidecars already have a parameter with namespace where the Lease objects are created, CSI driver operators
+      will need to set correctly in HyperShift, so CSI sidecars of different guest clusters don't collide.
+
+#### Notes about other CSI drivers
```

Comment:
What about CSI drivers from various on-prem storage vendors?  Would we need to maintain forks for each?

Reply:
We require them installed by OLM, i.e. they fully run in the guest cluster. Added a note about it to non-goals.